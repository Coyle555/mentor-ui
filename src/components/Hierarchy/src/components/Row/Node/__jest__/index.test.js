import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render, wait } from '@testing-library/react';
import { Node } from '../index';

afterEach(cleanup);

test('Default node render', () => {
	const tree = renderer.create(<Node />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Selected node render', () => {
	const tree = renderer.create(<Node selected={true} />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Node with a title', () => {
	const tree = renderer.create(<Node title="Foo" />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Node with a subtitle string', () => {
	const tree = renderer.create(<Node subtitle="Bar" />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Node with a subtitle function', () => {
	const node = { name: 'baz' };
	const subtitle = jest.fn(node => node.name);

	const tree = renderer.create(<Node node={node} subtitle={subtitle} />).toJSON();

	expect(tree).toMatchSnapshot();
	expect(subtitle).toHaveBeenCalledWith({ name: 'baz' });
});

test('Clicking a node', async () => {
	const node = { name: 'foo' };
	const onClick = jest.fn();

	const { container } = render(
		<Node node={node} onNodeClick={onClick} />);

	fireEvent.click(container.querySelector('div.mui-node-content'));
	expect(onClick).toHaveBeenCalledWith({ name: 'foo' });
});
