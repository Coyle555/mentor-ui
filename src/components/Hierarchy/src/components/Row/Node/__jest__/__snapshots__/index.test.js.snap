// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default node render 1`] = `
<div
  className="mui-node-content"
  onClick={[Function]}
>
  <span>
    <div
      className="node-text-title"
    >
      
    </div>
    <div
      className="node-text-subtitle"
    >
      
    </div>
  </span>
  <button
    className="node-buttons"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fal fa-ellipsis-v fa-3x"
    />
  </button>
</div>
`;

exports[`Node with a subtitle function 1`] = `
<div
  className="mui-node-content"
  onClick={[Function]}
>
  <span>
    <div
      className="node-text-title"
    >
      
    </div>
    <div
      className="node-text-subtitle"
    >
      baz
    </div>
  </span>
  <button
    className="node-buttons"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fal fa-ellipsis-v fa-3x"
    />
  </button>
</div>
`;

exports[`Node with a subtitle string 1`] = `
<div
  className="mui-node-content"
  onClick={[Function]}
>
  <span>
    <div
      className="node-text-title"
    >
      
    </div>
    <div
      className="node-text-subtitle"
    >
      Bar
    </div>
  </span>
  <button
    className="node-buttons"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fal fa-ellipsis-v fa-3x"
    />
  </button>
</div>
`;

exports[`Node with a title 1`] = `
<div
  className="mui-node-content"
  onClick={[Function]}
>
  <span>
    <div
      className="node-text-title"
    >
      Foo
    </div>
    <div
      className="node-text-subtitle"
    >
      
    </div>
  </span>
  <button
    className="node-buttons"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fal fa-ellipsis-v fa-3x"
    />
  </button>
</div>
`;

exports[`Selected node render 1`] = `
<div
  className="mui-node-content mui-node-selected"
  onClick={[Function]}
>
  <span>
    <div
      className="node-text-title"
    >
      
    </div>
    <div
      className="node-text-subtitle"
    >
      
    </div>
  </span>
  <button
    className="node-buttons"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fal fa-ellipsis-v fa-3x"
    />
  </button>
</div>
`;
